pandas/tseries/index.py

```diff
@@ -1769,14 +1772,17 @@ def indexer_between_time(self, start_time, end_time, include_start=True,
         -------
         values_between_time : TimeSeries
         """
-        from dateutil.parser import parse
+        if not isinstance(start_time, compat.string_types + (time,)) or \
```

Comment:
I don't think you need this check as long as you pass ``errors='raise'`` to ``to_time``